test/units/testsuite-21.sh

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+# vi: ts=4 sw=4 tw=0 et:
+
+set -eux
+set -o pipefail
+
+: >/failed
+
+RUN_OUT="$(mktemp)"
+
+run() {
+    "$@" |& tee "$RUN_OUT"
+}
+
+### SETUP ###
+# Configure network
+hostnamectl hostname ns1.unsigned.test
+echo "10.0.0.1 ns1.unsigned.test" >>/etc/hosts
+
+mkdir -p /etc/systemd/network
+cat >/etc/systemd/network/dns0.netdev <<EOF
+[NetDev]
+Name=dns0
+Kind=dummy
+EOF
+cat >/etc/systemd/network/dns0.network <<EOF
+[Match]
+Name=dns0
+
+[Network]
+Address=10.0.0.1/24
+DNSSEC=allow-downgrade
+DNS=10.0.0.1
+EOF
+
+{
+    echo "FallbackDNS="
+    echo "DNSSEC=allow-downgrade"
+    echo "DNSOverTLS=opportunistic"
+} >>/etc/systemd/resolved.conf
+ln -svf /run/systemd/resolve/stub-resolv.conf /etc/resolv.conf
+# Override the default NTA list, which turns off DNSSEC validation for (among
+# others) the test. domain
+mkdir -p "/etc/dnssec-trust-anchors.d/"
+echo local >/etc/dnssec-trust-anchors.d/local.negative
+
+# Sign the root zone
+keymgr . generate algorithm=ECDSAP256SHA256 ksk=yes zsk=yes
+# Create a trust anchor for resolved with our root zone
+keymgr . dnskey | sed 's/ DNSKEY/ IN DNSKEY/g' >/etc/dnssec-trust-anchors.d/root.positive
```

Comment:
I may asked that before, but why do you use DNSKEY instead of DS for positive trust anchors?
I am not familiar, but from dnssec-trust-anchors.d(5) says the following.
> It is generally recommended to encode trust anchors in DS resource records, rather than DNSKEY resource records.

Still I do not know if it is expected result, but if I create a positive trust anchor file with DNSKEY, then `dig DNSKEY .` provides only DNSKEY, no RRSIG.
If I create a positive trust anchor file with DS, then `dig DNSKEY .` provides both DNSKEY and RRSIG.

Reply:
> I may asked that before, but why do you use DNSKEY instead of DS for positive trust anchors? I am not familiar, but from dnssec-trust-anchors.d(5) says the following.
> 
> > It is generally recommended to encode trust anchors in DS resource records, rather than DNSKEY resource records.

*sigh*, I wish I has an answer for that, but I simply don't know - I vaguely remember I switched it to DS at some point, but I guess I didn't commit that change. With DS records in both DNSSEC trust anchors and `bind.keys` files, the test finally passes, at least on my machine. Let's see how CIs like it.

Thanks a lot for looking into this!